From 48a1afeb399fa85a72c523780fa9ef5a76fe3f86 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Fri, 17 Mar 2017 17:14:12 +0800
Subject: [PATCH 1477/2241] dts: a3700: add CPU PM clock and OPP entry in CPU
 nodes

Armada 3700 CPU frequency uses generic OPP-v2 framework,
need to register OPP entry inside of CPU DT nodes.
Also add CPU PM clock in CPU nodes.

Change-Id: Ie33ed835909dc8d7862c9ea57a9087e098b1aa6b
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37537
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Evan Wang <xswang@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-372x.dtsi |  2 ++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi | 11 +++++++++++
 2 files changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-372x.dtsi b/arch/arm64/boot/dts/marvell/armada-372x.dtsi
index f292a00..ee6cb47 100644
--- a/arch/arm64/boot/dts/marvell/armada-372x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-372x.dtsi
@@ -56,6 +56,8 @@
 			device_type = "cpu";
 			compatible = "arm,cortex-a53","arm,armv8";
 			reg = <0x1>;
+			cks = <&cpu_pm_clk 0>;
+			operating-points-v2 = <&cluster0_opp>;
 			enable-method = "psci";
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 9fcc1ae..24c3a3b 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -65,6 +65,15 @@
 		gpio1 = &gpio_sb;
 	};
 
+	cluster0_opp: opp_table0 {
+		compatible = "operating-points-v2";
+		opp-shared;
+		opp@0 {
+			opp-hz = /bits/ 64 <200000000>;
+			clock-latency-ns = <50000>;
+		};
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -72,6 +81,8 @@
 			device_type = "cpu";
 			compatible = "arm,cortex-a53", "arm,armv8";
 			reg = <0>;
+			clocks = <&cpu_pm_clk 0>;
+			operating-points-v2 = <&cluster0_opp>;
 			enable-method = "psci";
 		};
 	};
-- 
2.7.4

